pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -676,10 +676,10 @@ func (s *store) createPermissions(sess *db.Session, roleID int64, resource, reso
 		permissions = append(permissions, p)
 	}
 
-	// If there are no missing actions for the resource, we don't need to insert any prior actions
+	// If there are no missing actions for the resource (for instance, in case of access level downgrade), we don't need to insert any prior actions
 	// we still want to add the action set
 	if len(missingActions) == 0 {
```

Comment:
so is this case different from having, missing actions?

Reply:
It's a subset of this. `missingActions` will be empty if the permission gets downgraded or removed completely. In case of a downgrade (eg, folder editor to folder viewer), we still want to store the new action set (eg, folder viewer). But in case of permission removal we don't want to store the action set (action set would be empty - `folders:`).

Should I update the comments to make it clearer?

Reply:
If you feel like either making a comment or adding a function to express it. It would help for why I would say